### ISSUE TITLE
Output

### ISSUE THREAD

**Issue by chungyau97** [kivancyuksel/cfpd#1]

**Comment by chungyau97** [comment]
I'm using other face detector.
Why does it still predict landmark that have been cover? 
![image](https://user-images.githubusercontent.com/33013947/55694772-3bad2b00-59e8-11e9-81ae-36be9e5c9d95.png)
![Capture](https://user-images.githubusercontent.com/33013947/55694852-96468700-59e8-11e9-9577-50ecf4db497d.PNG)
![image](https://user-images.githubusercontent.com/33013947/55694973-1bca3700-59e9-11e9-9cc2-ce73a3df7670.png)

**Comment by kivancyuksel** [comment]
The reason for that is because the model has fixed output length (16) and the [dataset](https://ibug.doc.ic.ac.uk/resources/facial-point-annotations/) that is used to train the models always provides all 68 landmarks. So during training, for each case (no matter a facial part was covered or not) there were ground-truth landmarks coordinates. The model did not learn what to do in case a facial part was covered.

Also, if I didn't miss any, the dataset does not have any images where the face is covered like the way you are doing it. Hence the accuracy of the model is worse as we can see from the images. At the end, it is all about the data you have for training and their labels (ground-truth landmark coordinates in this case) . If this is the desired behavior for you and you have a labeled data to teach it, I advise you to re-train the network using the training script that is provided in this repo.

**Comment by chungyau97** [comment]
> The reason for that is because the model has fixed output length (16) and the [dataset](https://ibug.doc.ic.ac.uk/resources/facial-point-annotations/) that is used to train the models always provides all 68 landmarks. So during training, for each case (no matter a facial part was covered or not) there were ground-truth landmarks coordinates. The model did not learn what to do in case a facial part was covered.
> 
> Also, if I didn't miss any, the dataset does not have any images where the face is covered like the way you are doing it. Hence the accuracy of the model is worse as we can see from the images. At the end, it is all about the data you have for training and their labels (ground-truth landmark coordinates in this case) . If this is the desired behavior for you and you have a labeled data to teach it, I advise you to re-train the network using the training script that is provided in this repo.

I'm using your pretrained model. I'm exploring how to detect whether the face is covered or not. Have you come across any model that don't predict landmark when being covered?

**Comment by kivancyuksel** [comment]
I am sorry but I have not (but of course this doesn't mean that it doesn't exists).

**Comment by kivancyuksel** [comment]
> > The reason for that is because the model has fixed output length (16) and the [dataset](https://ibug.doc.ic.ac.uk/resources/facial-point-annotations/) that is used to train the models always provides all 68 landmarks. So during training, for each case (no matter a facial part was covered or not) there were ground-truth landmarks coordinates. The model did not learn what to do in case a facial part was covered.
> > Also, if I didn't miss any, the dataset does not have any images where the face is covered like the way you are doing it. Hence the accuracy of the model is worse as we can see from the images. At the end, it is all about the data you have for training and their labels (ground-truth landmark coordinates in this case) . If this is the desired behavior for you and you have a labeled data to teach it, I advise you to re-train the network using the training script that is provided in this repo.
> 
> I'm using your pretrained model. I'm exploring how to detect whether the face is covered or not. Have you come across any model that don't predict landmark when being covered?

I am sorry but I have not (but of course this doesn't mean that it doesn't exists).

**Comment by kivancyuksel** [comment]
Closing the issue.
